src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = tail = new Node<>(null, value, null);
+        } else {
+            Node<T> node = new Node<>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (head == null && tail == null) {
+            head = tail = new Node<>(null, value, null);
+        } else if (index == 0) {
+            Node<T> node = new Node<>(null, value, head);
+            head.prev = node;
+            head = node;
+        } else if (index == size) {
+            Node<T> node = new Node<>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        } else {
+            Node<T> current = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(current.prev, value, current);
+            current.prev.next = newNode;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = getNodeByIndex(index);
+        T result = current.value;
+        current.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNodeByIndex(index);
+        unlink(node);
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = getIndexByNode(object);
+        if (index == -1) {
+            return false;
+        } else {
+            unlink(getNodeByIndex(index));
+            return true;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return head == null;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> node = head;
+        for (int i = 0; i != index; i++) {
+            node = node.next;
+        }
+        return node;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
```

Comment:
make message more informative 
```suggestion
            throw new IndexOutOfBoundsException("Index: " + index + " out of bounds");
```